db-engine/src/main/java/com/splicemachine/db/iapi/types/SQLChar.java

```diff
@@ -927,6 +931,8 @@ private boolean evaluateNull()
     */
     public void writeExternal(ObjectOutput out) throws IOException {
         out.writeBoolean(isNull);
+        out.writeInt(sqlCharSize);
```

Comment:
My main concern of this fix is the Serde change for a basic data type. This may break the backward compatibility. For example, column stats info in system table may not be read correctly. Not sure if other data dictionary tables are affected.
If it is just column stats table that is affected, and the fix is going to a major release like 3.0, maybe it is OK, assuming a re-collection of stats is acceptable for a major release upgrade.

Reply:
Yes, it might be the case... As per my understanding: We can remove the Ser/De part, to retain backward compatibility. For this fix, as long as OLAP reinitializes the `Activation`, which is the current design, the SQLChar size information will be repopulated.